clippy_lints/src/utils/paths.rs

```diff
@@ -60,11 +60,11 @@ pub const OPTION_NONE: [&str; 4] = ["core", "option", "Option", "None"];
 pub const OPTION_SOME: [&str; 4] = ["core", "option", "Option", "Some"];
 pub const ORD: [&str; 3] = ["core", "cmp", "Ord"];
 pub const OS_STRING: [&str; 4] = ["std", "ffi", "os_str", "OsString"];
-pub const OS_STR_TO_OS_STRING: [&str; 5] = ["std", "ffi", "os_str", "OsStr", "to_os_string"];
+pub const OS_STR_TO_OS_STRING: [&str; 2] = ["<std::ffi::OsStr>", "to_os_string"];
 pub const PARTIAL_ORD: [&str; 3] = ["core", "cmp", "PartialOrd"];
 pub const PATH: [&str; 3] = ["std", "path", "Path"];
 pub const PATH_BUF: [&str; 3] = ["std", "path", "PathBuf"];
-pub const PATH_TO_PATH_BUF: [&str; 4] = ["std", "path", "Path", "to_path_buf"];
+pub const PATH_TO_PATH_BUF: [&str; 2] = ["<std::path::Path>", "to_path_buf"];
```

Comment:
These don't look good... That means `path_qualified` isn't quite right.

You want at least this, at the start of `path_qualified`:
```rust
        if trait_ref.is_none() {
            if let ty::Adt(def, substs) = self_ty.sty {
                return self.print_def_path(def.did, substs);
            }
        }
```

Reply:
Yeah that would be better. Should I also open a rustc PR to also change it there?

Reply:
Where? `rustdoc`? Maybe, I don't know what that's used for (cc @QuietMisdreavus)